### PR TITLE
Upgrade minio-go and add region and virtual host config for segcore c…

### PR DESCRIPTION
**Pull request by zhagnlu**
#25713

### CODE REVIEW

**Comment by yanliang567** [comment]
TBD after 2.2.12 released

**Comment by zhagnlu** [comment]
rerun ut

**Comment by zhagnlu** [comment]
rerun ut

**Comment by zhagnlu** [comment]
rerun ut

File: internal/core/src/storage/MinioChunkManager.cpp [hunk 1]
```diff
@@ -257,8 +262,8 @@ MinioChunkManager::MinioChunkManager(const StorageConfig& storage_config)
 }
 
 MinioChunkManager::~MinioChunkManager() {
-    ShutdownSDKAPI();
     client_.reset();
+    ShutdownSDKAPI();
```
Comment (xige-16) [inline internal/core/src/storage/MinioChunkManager.cpp:266]: why shutdown sdk after reseting client?

File: internal/core/unittest/test_minio_chunk_manager.cpp [hunk 2]
```diff
@@ -108,15 +110,16 @@ TEST_F(MinioChunkManagerTest, ObjectExist) {
...
-    if (!chunk_manager_->BucketExists(testBucketName)) {
-        chunk_manager_->CreateBucket(testBucketName);
-    }
+    // if (!chunk_manager_->BucketExists(testBucketName)) {
+    //     chunk_manager_->CreateBucket(testBucketName);
+    // }
+    auto has_bucket = chunk_manager_->BucketExists(testBucketName);
     uint8_t data[5] = {0x17, 0x32, 0x45, 0x34, 0x23};
-    string path = "1/3/5";
+    string path = "1";
```
Comment (xige-16) [inline internal/core/unittest/test_minio_chunk_manager.cpp:122]: This path seems to have nothing to do with pr, why should it be modified？

File: internal/core/src/storage/MinioChunkManager.cpp (hunk 1 above)
Comment (zhagnlu) [reply internal/core/src/storage/MinioChunkManager.cpp:266]: SDK API is high level than client, so need to close SDKAPI before client at the end.

**Comment by zhagnlu** [comment]
rerun ut

**Review by haorenfsa — APPROVED** [review]

File: internal/core/unittest/test_minio_chunk_manager.cpp (hunk 2 above)
Comment (zhagnlu) [reply internal/core/unittest/test_minio_chunk_manager.cpp:122]: can ignore this test, both is ok.

**Pull request merged**